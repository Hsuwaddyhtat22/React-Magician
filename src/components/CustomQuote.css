.custom-container {
  margin-top: 50px;
  margin-left: 50px;
}

.custom-wrapper {
  text-align: center;
}

.custom-quote {
  font-family: Arial, sans-serif;
  font-weight: bold;
  margin-left: 50px;
}

.custom-author {
  font-style: italic;
  color: #dcac00;
}

.app-heading {
  font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  font-size: 50px;
  color: #333;
  margin-left: 20px;
  margin-bottom: 20px;
}
