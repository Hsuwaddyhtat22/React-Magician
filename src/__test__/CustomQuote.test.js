import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import CustomQuote from '../components/QuoteDisplay';

// Mock the fetch function to return a resolved or rejected promise
global.fetch = jest.fn();

describe('CustomQuote component', () => {
  beforeEach(() => {
    global.fetch.mockClear();
  });

  it('renders loading text initially', async () => {
    global.fetch.mockResolvedValueOnce({
      json: async () => [],
    });

    render(<CustomQuote />);

    const loadingElement = screen.getByText('Loading...');
    expect(loadingElement).toBeInTheDocument();

    await waitFor(() => {
      expect(screen.queryByText('Loading...')).toBeNull();
    });

    expect(screen.getByTestId('quote-loading')).toMatchSnapshot();
  });

  it('renders quote and author after fetching data', async () => {
    global.fetch.mockResolvedValueOnce({
      json: async () => [{ quote: 'Test Quote', author: 'Test Author' }],
    });

    render(<CustomQuote />);

    // Wait for the fetch to resolve
    await waitFor(() => {
      const quoteElement = screen.getByText('Test Quote');
      const authorElement = screen.getByText('Test Author');
      expect(quoteElement).toBeInTheDocument();
      expect(authorElement).toBeInTheDocument();
    });

    expect(screen.getByTestId('quote-loaded')).toMatchSnapshot();
  });
});
