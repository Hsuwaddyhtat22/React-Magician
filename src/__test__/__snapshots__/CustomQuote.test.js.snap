// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CustomQuote component 1`] = `
<div>
  <div
    class="custom-container"
  >
    <div
      class="custom-wrapper"
    >
      <h2>
         
         
        Loading...
         
         
         
      </h2>
    </div>
  </div>
</div>
`;
